packages/compass-welcome/src/components/web-welcome-tab.tsx

```diff
@@ -26,8 +33,229 @@ const contentBodyStyles = css({
+});
+
+function ConnectionStatus({
+  connectionId,
+  isConnected,
+}: {
+  connectionId: string;
+  isConnected: boolean;
+}) {
+  const connectionInfo = useConnectionInfoForId(connectionId);
+  const connectingSteps = useConnectionConnectingSteps(connectionId);
+
+  if (!connectionInfo) {
+    return null;
+  }
+
+  const connectionName = connectionInfo.title;
+
+  // Create sub-items for connecting steps
+  const renderConnectingSteps = () => {
+    if (!connectingSteps) return null;
+
+    const steps = [
+      {
+        key: 'topology',
+        label: 'Discovering Topology...',
+        status: connectingSteps.topology,
+      },
+      {
+        key: 'authentication',
+        label: 'Authenticating...',
+        status: connectingSteps.authentication,
+      },
+      {
+        key: 'listingDatabases',
+        label: 'Listing Databases...',
+        status: connectingSteps.listingDatabases,
+      },
+    ];
+
+    return steps.map(({ key, label, status }) => {
+      const isCompleted = status === 'completed';
+      const isInProgress = status === 'in-progress';
+      const isFailed = status === 'failed';
+
+      return (
+        <div key={key} className={subItemStyles}>
+          {isCompleted ? (
+            <Icon glyph="Checkmark" size="small" className={greenIconStyles} />
+          ) : isFailed ? (
+            <Icon glyph="X" size="small" style={{ color: palette.red.base }} />
+          ) : isInProgress ? (
+            <SpinLoader size={10} />
+          ) : (
+            <div style={{ width: '10px', height: '10px' }} /> // Empty space for pending
+          )}
+          <span
+            style={{
+              color: isCompleted
+                ? palette.green.dark2
+                : isFailed
+                ? palette.red.base
+                : undefined,
+            }}
+          >
+            {label}
+          </span>
+        </div>
+      );
+    });
+  };
+
+  return (
+    <div>
+      <div
+        className={
+          isConnected
+            ? connectedConnectionItemStyles
+            : connectingConnectionItemStyles
+        }
+      >
+        {isConnected ? (
+          <Icon glyph="Checkmark" size="small" className={greenIconStyles} />
+        ) : (
+          <SpinLoader size={12} />
+        )}
+        <span>
+          {isConnected
+            ? `Connected to ${connectionName}`
+            : `Connecting to ${connectionName}`}
+        </span>
+      </div>
+      {!isConnected && <div>{renderConnectingSteps()}</div>}
+    </div>
+  );
+}
+
 export default function WebWelcomeTab() {
   const numConnections = useConnectionIds().length;
+
+  // Get IDs of connections that are currently connecting
+  const connectingConnectionIds = useConnectionIds(
+    (connection) => connection.status === 'connecting'
+  );
+
+  // Get IDs of connections that are connected
+  const connectedConnectionIds = useConnectionIds(
+    (connection) => connection.status === 'connected'
+  );
```

Comment:
This should probably include errored connections also, otherwise they jump in and out of this view in a very weird way